pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -82,11 +84,28 @@ func (service *IPAddressAllocationService) CreateOrUpdateIPAddressAllocation(obj
 		return false, err
 	}
 	log.V(1).Info("Existing ipaddressallocation", "ipaddressallocation", existingIPAddressAllocation)
+
+	if existingIPAddressAllocation != nil && existingIPAddressAllocation.AllocationIps != nil && existingIPAddressAllocation.AllocationSize == nil {
+		// For the restored NSX VPC IPAddressAllocation, its allocation_size is null.
+		// We cannot update it in the normal reconcile after the restore, because the error:
+		// "nsx error code: 612866, message: Properties IP block, allocation IPs, allocation IP, IP block visibility, allocation size and IPAddressType of existing Vpc IP address allocation: <VPC IP address allocation path> can not be modified."
```

Comment:
When new tag is added to the restored IPAddressAllocation, will the update be triggered and resulted in this error?
Shall we propagate the AllocationIps from existingIPAddressAllocation to nsxIPAddressAllocation and remove AllocationSize? Or just updating the tags field of the existingIPAddressAllocation?

Reply:
I updated the logic, now we remove the `AllocationSize` from desired `nsxIPAddressAllocation` instead of filling `AllocationSize` to `existingIPAddressAllocation`, I think this can cover the case that the properties like `tags`, `display_name` need to be updated. I think we don't need to propagate the `AllocationIps` from `existingIPAddressAllocation` to `nsxIPAddressAllocation` because the error 612866 indicates that the `AllocationIps` is immutable and we don't compare this field in the comparison function.

Reply:
I refer to this case: When we update tags, we will call the patch API with `nsxIPAddressAllocation`. And this `nsxIPAddressAllocation` generated by `service.BuildIPAddressAllocation(obj)` in the normal reconcile does not contains the field AllocationIps. But I think if the previous IPAddressAllocation is restored with AllocationIps, and we update it without AllocationIps will lead to this error 612866?
I'm not sure if I have missed something. Have you tested the case update tags for a restored IPAddressAllocation?

Reply:
Many thanks for your good caught! It's indeed an issue, I updated the code to propagate the `AllocationIps` from `existingIPAddressAllocation` to `nsxIPAddressAllocation` as a fix and adding the test steps in case A, step 5-6, please check.